src/test/java/io/cryostat/resources/LocalStackResource.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.resources;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.test.common.DevServicesContext;
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer.Service;
+import org.testcontainers.utility.DockerImageName;
+
+public class LocalStackResource
```

Comment:
I think this is already handled by the devservices part of the Quarkus S3 extension:

https://github.com/cryostatio/cryostat3/blob/f4568124b7857df9a431f2135b10bc0c3db6d961/pom.xml#L149

https://github.com/cryostatio/cryostat3/blob/f4568124b7857df9a431f2135b10bc0c3db6d961/src/main/resources/application-test.properties#L23

I was already starting on a similar idea here however to replace this devservices config with a QuarkusTestResource, with the goal of making it configurable so the tests can plug in different kinds of S3 instead of always using Localstack.

So I think keeping this new LocalStackResource is OK, but the `application-test.properties` should be examined and see if the devservices configurations should be removed - it seems like they would conflict with this, or perhaps just effectively override it.

Reply:
Without it, in the IT it fails with connection refused (the hostname and port seems wrong). 
In the UT, it fails with aws credentials missing. I tried adding the credentials to the properties files, but without success. It seems that it does not work on the tests.

Reply:
https://github.com/lkonno/cryostat3/pull/1

Reply:
Still not quite working here, the unit tests complain that the expected bucket (`archivedrecordings`) doesn't exist. But it looks like I have at least got localstack deploying using the QuarkusTestResource, and it looks like Cryostat3 is able to communicate with this localstack instance. Cryostat3 is supposed to check if the expected buckets exist and create them if they don't exist whenever the Cryostat3 server starts:

https://github.com/cryostatio/cryostat3/blob/f4568124b7857df9a431f2135b10bc0c3db6d961/src/main/java/io/cryostat/recordings/Recordings.java#L136

so I guess this is somehow either not getting triggered, or some other part of the configuration is wrong or perhaps changing during the test execution.

Reply:
It looks like the updated ITs work with that change too, so it is now just down to figuring out how to get the S3 buckets to be created properly so that the Recordings class can manage the archives properly.

Reply:
These unit test startup logs look relevant:

```
2023-11-14 13:09:18,395 INFO  [io.cry.dis.JDPDiscovery] (main) Starting JDP client
2023-11-14 13:09:18,396 INFO  [io.cry.cor.log.Logger] (main) JDP Discovery started
2023-11-14 13:09:18,919 INFO  [io.cry.rec.Recordings] (main) software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 500, Request ID: b27bbc54-e9ac-4b03-a08e-25a42d9809dd, Extended Request ID: s9lzHYrFp76ZVxRcpX9+5cjAnEH2ROuNkd2BHfIa6UkFVdtjf5mKR3/eTPFvsiP/XV/VLi31234=)
2023-11-14 13:09:19,291 ERROR [io.cry.rec.Recordings] (main) software.amazon.awssdk.services.s3.model.S3Exception: exception while calling s3 with unknown operation: Unable to find operation for request to service s3: PUT / (Service: S3, Status Code: 500, Request ID: ce639200-8444-402b-b175-caf038becd62, Extended Request ID: s9lzHYrFp76ZVxRcpX9+5cjAnEH2ROuNkd2BHfIa6UkFVdtjf5mKR3/eTPFvsiP/XV/VLi31234=)
```

Reply:
Fixed it, just needed to enable path-style S3 access (as opposed to subdomain access). This is already needed in the smoketest environment and it's needed in the unit/integration test harness for the same reasons.

The recording workflow test is still failing but it progresses further.

Reply:
Oh - it fails expecting an HTML formatted report, so actually it looks like it's probably OK :-) just needs to be adjusted accordingly for the JSON-only API change.

Reply:
Updated the test to reflect what the equivalent 2.4 test actually expects as far as a JSON report response - these integration tests were ported over before that change had happened in the old repo, so they're a little desynced and out of date...

Now I have this:

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 50.28 s <<< FAILURE! -- in itest.RecordingWorkflowTest
[ERROR] itest.RecordingWorkflowTest.testWorkflow -- Time elapsed: 7.471 s <<< FAILURE!
java.lang.AssertionError: 

Expected: ""
     but: was "{HeapInspectionGc=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@356ee58e, ApplicationHalts=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@6e355d9a, ClassLeak=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@5df298a2, DiscouragedVmOptions=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@2143f12a, IncreasingMetaSpaceLiveSet=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@63c32219, VMOperations=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@275e6a76, TlabAllocationRatio=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@6ea0c1b7, OverAggressiveRecordingSetting=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@5054a7e8, FlightRecordingSupport=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@2a7d9411, ProcessStarted=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@7e2456f2, JavaBlocking=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@16e19df8, GcStall=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@637f032b, PasswordsInSystemProperties=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@25e4ce49, Allocations.class=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@79ab9fc1, ManagementAgent=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@633fd351, LongGcPause=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@78c2e246, StringDeduplication=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@7b57ed91, SystemGc=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@5ea730ad, FileWrite=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@439fc573, GcPauseRatio=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@2ca0df94, MetaspaceOom=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@68df84f4, Errors=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@f9af31a, Fatal Errors=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@32131bc7, biasedLockingRevocation=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@18653e6e, ClassLoading=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@c09e1b2, ContextSwitch=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@54d3d10a, FewSampledThreads=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@7eb43574, BytecodeVerification=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@4f13f282, Options=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@cd49ea3, GcLocker=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@464545b2, CodeCache=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@1905d79b, GcFreedRatio=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@49d2a1db, BufferLost=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@615b1fc4, PasswordsInEnvironment=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@ddfae88, GcOptions=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@3ad109a7, DMSIncident=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@6651ab5f, MethodProfiling=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@5f4d3d6e, biasedLockingRevocationPause=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@2f9663e1, LowOnPhysicalMemory=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@1db8db85, FileRead=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@7658803e, ManyRunningProcesses=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@404875d6, JfrPeriodicEventsFix=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@5cdd9f55, SocketWrite=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@76203b8, GarbageCollectionInfoRule=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@35093a1a, Exceptions=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@36ec0fa8, CompareCpu=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@67b0917, StackdepthSetting=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@1193b5a3, HeapDump=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@6db69461, Allocations.thread=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@3bf14730, HeapContent=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@651cd2e8, DebugNonSafepoints=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@692e4d45, SocketRead=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@30e2c99d, PrimitiveToObjectConversion=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@11bd076, HighGc=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@213d92f, CompressedOops=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@194cf84d, DumpReason=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@3f4474d8, PasswordsInArguments=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@60839b26, IncreasingLiveSet=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@7cc5fcb8, HighJvmCpu=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@235edfd, DuplicateFlags=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@1e4bbbd7, FullGc=io.cryostat.core.reports.InterruptibleReportGenerator$AnalysisResult@62ed5993}"
```

Looks like the JSON response is malformed, the AnalyisResult type does not get serialized properly.

I'm going to go ahead and close https://github.com/lkonno/cryostat3/pull/1 , just pushing work directly to this PR branch instead.